robottelo/config/settings.py

```diff
@@ -0,0 +1,365 @@
+            'performance', 'time_hammer', False, bool)
+        self.performance.cdn_address = self.reader.get(
+            'performance', 'cdn_address')
+        self.performance.virtual_machines = self.reader.get(
+            'performance', 'virtual_machines', cast=list)
+        self.performance.fresh_install_savepoint = self.reader.get(
+            'performance', 'fresh_install_savepoint')
+        self.performance.enabled_repos_savepoint = self.reader.get(
+            'performance', 'enabled_repos_savepoint')
+        self.performance.csv_buckets_count = self.reader.get(
+            'performance', 'csv_buckets_count', 10, int)
+        self.performance.sync_count = self.reader.get(
+            'performance', 'sync_count', 3, int)
+        self.performance.sync_type = self.reader.get(
+            'performance', 'sync_type', 'sync')
+        self.performance.repos = self.reader.get(
+            'performance', 'repos', cast=list)
+
+        if self.performance.cdn_address is None:
+            self._validation_errors.append(
+                '[performance] cdn_address must be provided.')
+        if self.performance.virtual_machines is None:
+            self._validation_errors.append(
+                '[performance] virtual_machines must be provided.')
+        if self.performance.fresh_install_savepoint is None:
+            self._validation_errors.append(
+                '[performance] fresh_install_savepoint must be provided.')
+        if self.performance.enabled_repos_savepoint is None:
+            self._validation_errors.append(
+                '[performance] enabled_repos_savepoint must be provided.')
+
+    @property
+    def configured(self):
+        """Returns True if the settings have already been configured."""
+        return self._configured
+
+    def _configure_logging(self):
+        """Configure logging for the entire framework.
+
+        If a config named ``logging.conf`` exists in Robottelo's root
+        directory, the logger is configured using the options in that file.
+        Otherwise, a custom logging output format is set, and default values
+        are used for all other logging options.
+        """
+        # All output should be made by the logging module, including warnings
+        logging.captureWarnings(True)
+
+        # Set the logging level based on the Robottelo's verbosity
+        for name in ('nailgun', 'robottelo'):
+            logging.getLogger(name).setLevel(self.verbosity)
+
+        # Allow overriding logging config based on the presence of logging.conf
+        # file on Robottelo's project root
+        logging_conf_path = os.path.join(get_project_root(), 'logging.conf')
+        if os.path.isfile(logging_conf_path):
+            config.fileConfig(logging_conf_path)
+        else:
+            logging.basicConfig(
+                format='%(levelname)s %(module)s:%(lineno)d: %(message)s'
+            )
+
+    def _configure_third_party_logging(self):
```

Comment:
Better to leave it inside the class to have the context of the configuration. This should be used only here and not outside. Also I'm thinking on dropping this because we should now change the behavior of third party logging.